app/src/main/java/com/codecool/lolapp/adapters/ListingAdapter.kt

```diff
@@ -0,0 +1,84 @@
+package com.codecool.lolapp.adapters
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ImageButton
+import android.widget.ImageView
+import android.widget.TextView
+import androidx.appcompat.app.AppCompatActivity
+import androidx.recyclerview.widget.RecyclerView
+import com.bumptech.glide.Glide
+import com.bumptech.glide.request.RequestOptions
+import com.codecool.lolapp.R
+import com.codecool.lolapp.fragments.DetailsFragment
+import com.codecool.lolapp.model.Character
+import com.codecool.lolapp.util.BASE_URL_IMAGE
+import com.codecool.lolapp.util.BASE_URL_TYPE
+
+
+class ListingAdapter(var characters: ArrayList<Character>) :
+    RecyclerView.Adapter<ListingAdapter.ListingViewHolder>() {
+
+    fun updateCharacters(newCharacters: List<Character>) {
+        characters.clear()
+        characters.addAll(newCharacters)
+        notifyDataSetChanged()
+    }
+
+    class ListingViewHolder(private val view: View) : RecyclerView.ViewHolder(view) {
+        private val image = view.findViewById<ImageView>(R.id.character_image)
+        private val name = view.findViewById<TextView>(R.id.character_name)
+        private val title = view.findViewById<TextView>(R.id.character_title)
+        private val blurb = view.findViewById<TextView>(R.id.character_blurb)
+        private val detailsButton = view.findViewById<ImageButton>(R.id.buttonDetails)
+
+        fun bind(character: Character) {
+            if (image != null) {
+                val url = BASE_URL_IMAGE + character.name + BASE_URL_TYPE
```

Comment:
try having a separate class to map the image URL into the final form, this should not be the responsibility of the view
maybe after you download it in the presenter.
I know it was last minute and that's the reason why it stayed like this, I just wanted to share my advice anyway